src/test/java/com/kasperin/inventory_management/controllers/v1/ProcessedFoodControllerTest.java

```diff
@@ -0,0 +1,205 @@
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].id").value(1L))
+                .andExpect(jsonPath("$[1].id").value(2L));
+
+        verify(processedFoodService).findAll();
+        verify(processedFoodService, never()).findByType(any());
+    }
+
+    @Test
+    void findAllByType_vegan() throws Exception {
+        // given
+        ProcessedFood pf = new ProcessedFood();
+        pf.setId(1L);
+        pf.setFoodType(FoodType.VEGAN);
+
+        when(processedFoodService.findByType(any())).thenReturn(Collections.singletonList(pf));
+
+        // when
+        mockMvc.perform(get("/api/v1/processedFoods?type=VEGAN")
+                .accept(MediaType.APPLICATION_JSON))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].foodType").value(FoodType.VEGAN.name()));
+
+        verify(processedFoodService).findByType(eq(FoodType.VEGAN));
+        verify(processedFoodService, never()).findAll();
+    }
+
+    @Test
+    void findAllByType_nonvegan() throws Exception {
+        // given
+        ProcessedFood pf = new ProcessedFood();
+        pf.setId(1L);
+        pf.setFoodType(FoodType.NONVEGAN);
+
+        when(processedFoodService.findByType(any())).thenReturn(Collections.singletonList(pf));
+
+        // when
+        mockMvc.perform(get("/api/v1/processedFoods?type=NONVEGAN")
+                .accept(MediaType.APPLICATION_JSON))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].foodType").value(FoodType.NONVEGAN.name()));
+
+        verify(processedFoodService).findByType(eq(FoodType.NONVEGAN));
+        verify(processedFoodService, never()).findAll();
+    }
+
+    @Test
+    void findByName() throws Exception {
+        final Long id = 1L;
+        final String name = "name";
+
+        // given
+        ProcessedFood pf = new ProcessedFood();
+        pf.setId(id);
+        pf.setName(name);
+
+        when(processedFoodService.findByName(anyString())).thenReturn(Optional.of(pf));
+
+        // when
+        mockMvc.perform(get("/api/v1/processedFoods/by-name/name").accept(MediaType.APPLICATION_JSON))
+                // then
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name));
+
+        verify(processedFoodService).findByName(eq(name));
+    }
+
+    @Test
+    void findByName_notFound() throws Exception {
+        // given
+        when(processedFoodService.findByName(anyString())).thenReturn(Optional.empty());
+
+        // when
+        mockMvc.perform(get("/api/v1/processedFoods/by-name/wat").accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void findById() throws Exception {
+        // given
+        ProcessedFood pf = new ProcessedFood();
+        pf.setId(1L);
```

Comment:
Again, note I'm not filling out the properties. I just want to check that the code is executing the way I expect (with `.andExpect()` calls and `verify()`